doc/lsst.daf.butler/datastores.rst

```diff
@@ -0,0 +1,79 @@
+.. _daf_butler-datastores:
+
+#######################
+Datastore Configuration
+#######################
+
+A Butler `~lsst.daf.butler.Datastore` is configured in the ``datastore`` section of the top-level Butler YAML configuration.
+The only mandatory entry in the datastore configuration is the ``cls`` key.
+This specifies the fully qualified class name of the Python class implementing the datastore.
+The default is to use the `~lsst.daf.butler.datastores.posixDatastore.PosixDatastore`.
+All other keys depend on the specific datastore class that is selected.
+
+The default configuration values can be inspected at ``$DAF_BUTLER_DIR/config`` and current values can be obtained by calling ``butler config-dump`` on a butler repository.
+
+File-Based Datastores
+=====================
+
+The file-based datastores (for example `~lsst.daf.butler.datastores.posixDatastore.PosixDatastore` and `~lsst.daf.butler.datastores.s3Datastore.S3Datastore`) share configuration since they use formatters to serialize datasets to file artifacts using file name template schemes and also support composite disassembly.
+
+The supported configurations are:
+
```

Comment:
Should the following be a bulleted list?

Reply:
I thought a description list was more accurate for what this is. Granted there is no real formatting there that tells you it is a description list. Maybe I can boldify the headings.